File: go/acronym/acronym.go
package acronym

import "strings"


func Abbreviate(s string) string {

	s = strings.Replace(s, "-", " ", -1)
	f := strings.Fields(s)


	out := ""
	for _, c := range f {
		out += strings.ToUpper(string(c[0]))
	}

	return out
}

File: haskell/acronym/src/Acronym.hs
module Acronym (abbreviate) where

import Data.Char (isUpper, toUpper, isSpace, isAlpha)

abbreviate :: String -> String
abbreviate "" = ""
abbreviate (x:y:xs)
    | x `elem` "- "  && isAlpha y = toUpper y : abbreviate xs --Spaces or hyphens
    | isUpper x && isUpper y = abbreviate (x:xs) --throw out Y
    | isUpper x              = x:abbreviate (xs)
    | otherwise              = abbreviate (y:xs)
abbreviate (x:xs)
    | isUpper x              = x:abbreviate xs
    | otherwise              = abbreviate xs